airbyte-integrations/connectors/source-intercom/source_intercom/components.py

```diff
@@ -361,14 +365,7 @@ def wrapper_balance_rate_limit(*args, **kwargs):
 
 
 @dataclass(eq=False)
-class HttpRequesterWithRateLimiter(HttpRequester):
-    """
-    The difference between the built-in `HttpRequester` and this one is the custom decorator,
-    applied on top of `interpret_response_status` to preserve the api calls for a defined amount of time,
-    calculated using the rate limit headers and not use the custom backoff strategy,
-    since we deal with Response.status_code == 200,
-    the default requester's logic doesn't allow to handle the status of 200 with `should_retry()`.
-    """
+class IntercomHttpRequester(HttpRequester):
```

Comment:
Something that seems unclear to me is where this custom `IntercomHttpRequester` differs from the regular `HttpRequester` that is supplied by the low-code CDK, since we removed `interpret_response_status` into a custom error handler, it doesn't seem like this class is overriding anything meaningful.

I see that we instantiate a bunch of interpolators, but the overwritten `get_request_params()` and other methods seem to just do the same behavior as [interpolated_request_options_provider.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/interpolated_request_options_provider.py) which makes me think this class is not needed anymore.

What do you think @darynaishchenko ? If this is in fact needed, let's add a docstring explaining the need for this custom class. Otherwise we should just swap this out and use `HttpRequester` in the manifest

Reply:
Yes, you are right, here is no need to overwrite `HttpRequester` now. Changed it manifest file. 
Run regression tests [here](https://github.com/airbytehq/airbyte/actions/runs/10110305152/job/27959956316): request urls looks good.